homeassistant/components/homee/__init__.py

```diff
@@ -0,0 +1,274 @@
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: HomeeConfigEntry) -> bool:
+    """Set up homee from a config entry."""
+    # Create the Homee api object using host, user,
+    # password & pyHomee instance from the config
+    homee = Homee(
+        host=entry.data[CONF_HOST],
+        user=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        device="pymee_" + hass.config.location_name,
+        reconnect_interval=10,
+        max_retries=100,
+    )
+
+    # Start the homee websocket connection as a new task
+    # and wait until we are connected
+    hass.loop.create_task(homee.run())
+    await homee.wait_until_connected()
+
+    # Log info about nodes, to facilitate recognition of unknown nodes.
+    for node in homee.nodes:
+        _LOGGER.info(
+            "Found node %s, with following Data: %s",
+            node.name,
+            node.raw_data,
+        )
+
+    entry.runtime_data = HomeeRuntimeData(homee)
+
+    # create device register entry
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        connections={
+            (dr.CONNECTION_NETWORK_MAC, dr.format_mac(homee.settings.mac_address))
+        },
+        identifiers={(DOMAIN, homee.settings.uid)},
+        manufacturer="homee",
+        name=homee.settings.homee_name,
+        model="homee",
+        sw_version=homee.settings.version,
+        hw_version="TBD",
+    )
+
+    # Forward entry setup to the platforms
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    entry.async_on_unload(entry.add_update_listener(async_update_entry))
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: HomeeConfigEntry) -> bool:
+    """Unload a homee config entry."""
+    # Unload platforms
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        # Get Homee object and remove it from data
+        homee: Homee = entry.runtime_data.homee
+
+        # Schedule homee disconnect
+        homee.disconnect()
+
+    return unload_ok
+
+
+async def async_update_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reload homee integration after config change."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: HomeeConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    homee = config_entry.runtime_data.homee
+    model = NodeProfile[device_entry.model].value
+    for node in homee.nodes:
+        # 'identifiers' is a set of tuples, so we need to check for the tuple.
+        if ("homee", node.id) in device_entry.identifiers and node.profile == model:
+            # If Node is still present in Homee, don't delete.
+            return False
+
+    return True
+
+
+class HomeeNodeEntity:
```

Comment:
This looks like a base entity, rather move it to `entity.py`

Reply:
Shouldn't this also inherit from ```Entity```?

Reply:
yes